pkg/resources/server.go

```diff
@@ -291,10 +323,28 @@ func (rs *resourceServer) ListAndWatch(e *pluginapi.Empty, s pluginapi.DevicePlu
 				rs.updateResource <- true
 				return err
 			}
+			err := rs.updateCDISpec()
```

Comment:
if you do it here, you need to protect using rs.mutex like done in rs.sendDevices()